### PR TITLE
React Wrapper Step 3: Split page.html.ejs along column divide

### PR DESCRIPTION
**Pull request by islemaster**
Splits the page.html.ejs template into two files, one for the contents of `#visualizationColumn` and one for the contents of `#codeWorkspace`.  Passes a separate render function for each section to the top-level React component.

This has several side effects:
- I've separated out the properties passed to visualizationColumn.ejs vs codeWorkspace.ejs so they only receive what they need.  This had to be done on an app-by-app basis since they each render a little differently.
  - In most cases there's a nice clean separation, which will help with further refactoring down the road.
  - applab and gamelab have the worst entanglement / props needed in both views.  I think the debugger might be to blame.
- We break out our first unique top-level component `NetSimView` because NetSim does not have a `#visualizationColumn` or a `#codeWorkspace` so it can't participate in this transform.
- Some common boilerplate between `AppView` and `NetSimView` gets factored into `StudioAppWrapper`, including the rule for whether to render `RotateContainer` ([see related conversation](https://github.com/code-dot-org/code-dot-org/pull/7026#issuecomment-190387459) on #7026).
- `ProtectedStatefulDiv` now takes an `id` property (among others), since we have lots of code that depends on access to `#visualizationColumn` and `#codeWorkspace`.
- `#visualizationResizeBar` is now rendered entirely by React, though it's also locked and will not update since we manually bind handlers against it outside of React.

_Step 3 from [this design document](https://docs.google.com/document/d/1itHwWa0ilE4FAMjDuqxkWXSqp_d0z7lhTeMFJybFH0w/edit?usp=sharing)._
# Testing

Worst bug surface in this change is probably from typos in breaking out the properties passed to the EJS rendering functions.
- [x] Run UI tests locally via chromedriver
- [x] Run UI tests via Saucelabs against local
- [x] Run eyes tests via Saucelabs against local
- [x] Manually verify each app type in allthethings
# Progress

Pretty cool to see React slowly reach into the page:
screenshot from 2016-02-29 15-51-01

### CODE REVIEW

**Comment by islemaster** [comment]
CC @bcjordan @davidsbailey @Hamms @joshlory

File: apps/src/templates/codeWorkspace.html.ejs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      <div id="toolbox-header" class="workspace-header">
+        <i id="hide-toolbox-icon" style="display:none;" class="fa fa-chevron-circle-right"></i>
+        <span><%= data.editCode ? msg.toolboxHeaderDroplet() : msg.toolboxHeader() %></span>
+      </div>
+      <div id="show-toolbox-header" class="workspace-header" style="display:none;">
+        <i id="show-toolbox-icon" class="fa fa-chevron-circle-right"></i>
+        <span><%= msg.showToolbox() %></span>
+      </div>
+      <div id="show-code-header" class="workspace-header workspace-header-button"><span><img src="<%= assetUrl('media/applab/blocks_glyph.gif') %>" class="blocks-glyph" /><i class="fa fa-code"></i><%= msg.showCodeHeader() %></span></div>
+      <% if (!data.readonlyWorkspace) { %>
```
Comment (bcjordan) [inline apps/src/templates/codeWorkspace.html.ejs:17]: I noticed this PR removes indentation inside most ejs `if` blocks, intentional?

**Comment by bcjordan** [comment]
LGTM! Love the separation and naming

**Comment by Bjvanminnen** [comment]
lgtm. nice progress! :)

File: apps/src/templates/codeWorkspace.html.ejs (hunk 1 above)
Comment (islemaster) [inline apps/src/templates/codeWorkspace.html.ejs:17]: Ah, no, I missed that.  Autoreformat on paste in RubyMine rearing its (normally useful) head.  Any suggestions on fixing it's behavior in these sort of cases?

Comment (bcjordan) [inline apps/src/templates/codeWorkspace.html.ejs:17]: Wasn't able to find any options to fix this, looks like the JetBrains `ejs` plugin is just wrong.

image

Opened [a ticket](https://youtrack.jetbrains.com/issue/RUBY-17879) (don't see a GH repo for the integration)

**Pull request merged**